src/features/PowerShellNotebooks.ts

```diff
@@ -0,0 +1,220 @@
+                    notebookData.cells.push({
+                        cellKind: cellKind!,
+                        language: cellKind === vscode.CellKind.Markdown ? 'markdown' : 'powershell',
+                        outputs: [],
+                        source: curr.join("\n"),
+                        metadata: {
+                            custom: {
+                                commentType: cellKind === vscode.CellKind.Markdown ? CommentType.LineComment : CommentType.Disabled,
+                            }
+                        }
+                    });
+                }
+
+                // set initial new cell state
+                curr = [];
+                cellKind = kind;
+                curr.push(kind === vscode.CellKind.Markdown ? lines[i].replace(/^\#\s*/, '') : lines[i]);
+            }
+        }
+
+        if (curr.length) {
+            notebookData.cells.push({
+                cellKind: cellKind!,
+                language: cellKind === vscode.CellKind.Markdown ? 'markdown' : 'powershell',
+                outputs: [],
+                source: curr.join("\n"),
+                metadata: {
+                    custom: {
+                        commentType: cellKind === vscode.CellKind.Markdown ? CommentType.LineComment : CommentType.Disabled,
+                    }
+                }
+            });
+        }
+
+        return notebookData;
+    }
+
+    resolveNotebook(document: vscode.NotebookDocument, webview: { readonly onDidReceiveMessage: vscode.Event<any>; postMessage(message: any): Thenable<boolean>; asWebviewUri(localResource: vscode.Uri): vscode.Uri; }): Promise<void> {
+        return;
+    }
+
+    saveNotebook(document: vscode.NotebookDocument, cancellation: vscode.CancellationToken): Promise<void> {
+        return this._save(document, document.uri, cancellation);
+    }
+
+    saveNotebookAs(targetResource: vscode.Uri, document: vscode.NotebookDocument, cancellation: vscode.CancellationToken): Promise<void> {
+        return this._save(document, targetResource, cancellation);
+    }
+
+    async backupNotebook(document: vscode.NotebookDocument, context: vscode.NotebookDocumentBackupContext, cancellation: vscode.CancellationToken): Promise<vscode.NotebookDocumentBackup> {
+        await this._save(document, context.destination, cancellation);
+
+        return {
+            id: context.destination.toString(),
+            delete: () => {
+                vscode.workspace.fs.delete(context.destination);
+            }
+        };
+    }
+
+    public dispose() {
+        this.showNotebookModeCommand.dispose();
+        this.hideNotebookModeCommand.dispose();
+    }
+
+    public setLanguageClient(languageClient: LanguageClient) {
+        this.languageClient = languageClient;
+    }
+
+    async _save(document: vscode.NotebookDocument, targetResource: vscode.Uri, _token: vscode.CancellationToken): Promise<void> {
+        const retArr: string[] = [];
+        document.cells.forEach((cell) => {
```

Comment:
I notice `.forEach()` is used here but `for (... of ...)` is used below

Reply:
moved to `forof`